src/components/viewmodels/rooms/UserIdentityWarningViewModel.tsx

```diff
@@ -0,0 +1,164 @@
+    const [currentPrompt, setCurrentPrompt] = useState<ViolationPrompt | undefined>(undefined);
+
+    const loadViolations = useMemo(
+        () =>
+            throttle(async (): Promise<void> => {
+                const isEncrypted = crypto && (await crypto.isEncryptionEnabledInRoom(room.roomId));
+                if (!isEncrypted) {
+                    setMembers([]);
+                    setCurrentPrompt(undefined);
+                    return;
+                }
+
+                const targetMembers = await room.getEncryptionTargetMembers();
+                setMembers(targetMembers);
+                const violations = await mapToViolations(crypto, targetMembers);
+
+                let candidatePrompt: ViolationPrompt | undefined;
+                if (violations.length > 0) {
+                    // sort by user ID to ensure consistent ordering
+                    const sortedViolations = violations.sort((a, b) => a.member.userId.localeCompare(b.member.userId));
+                    candidatePrompt = sortedViolations[0];
+                } else {
+                    candidatePrompt = undefined;
+                }
+
+                // is the current prompt still valid?
+                setCurrentPrompt((existingPrompt): ViolationPrompt | undefined => {
+                    if (existingPrompt && violations.includes(existingPrompt)) {
+                        return existingPrompt;
+                    } else if (candidatePrompt) {
+                        return candidatePrompt;
+                    } else {
+                        return undefined;
+                    }
+                });
+            }),
+        [crypto, room],
+    );
+
+    // We need to listen for changes to the members list
+    const onRoomStateEvent = useCallback(
+        async (event: MatrixEvent): Promise<void> => {
+            if (!crypto || event.getRoomId() !== room.roomId) {
+                return;
+            }
+            let shouldRefresh = false;
+
+            const eventType = event.getType();
+
+            if (eventType === EventType.RoomEncryption && event.getStateKey() === "") {
+                // Room is now encrypted, so we can initialise the component.
+                shouldRefresh = true;
+            } else if (eventType == EventType.RoomMember) {
+                // We're processing an m.room.member event
+                // Something has changed in membership, someone joined or someone left or
+                // someone changed their display name. Anyhow let's refresh.
+                const userId = event.getStateKey();
+                shouldRefresh = !!userId;
+            }
+
+            if (shouldRefresh) {
+                loadViolations().catch((e) => {
+                    logger.error("Error refreshing UserIdentityWarningViewModel:", e);
+                });
+            }
+        },
+        [crypto, room, loadViolations],
+    );
+    useTypedEventEmitter(cli, RoomStateEvent.Events, onRoomStateEvent);
```

Comment:
For functions that are defined and then just used once on the next line, probably may as well just use an anonymous one inline.

Reply:
Done in https://github.com/element-hq/element-web/pull/29067/commits/6b76fbc39273ebef8366a2eb3492e6d87d8f9c67